pkg/controllers/disruption/controller.go

```diff
@@ -242,3 +248,36 @@ func (c *Controller) logAbnormalRuns(ctx context.Context) {
 		}
 	}
 }
+
+// logInvalidBudgets will log if there are any invalid schedules detected
+func (c *Controller) logInvalidBudgets(ctx context.Context) {
+	nodePoolList, err := nodepoolutil.List(ctx, c.kubeClient)
+	if err != nil {
+		logging.FromContext(ctx).Debugf("listing nodepools, %s", err)
+	}
+	var buf bytes.Buffer
+	for _, np := range nodePoolList.Items {
+		for _, budget := range np.Spec.Disruption.Budgets {
```

Comment:
Seems like a lot of parsing the same thing over again. Is there a way to unify this code into `v1beta1` so that we are using duplications of the code (one just returns an error while the other checks for the presence of the error and fails closed?)